staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -215,6 +215,10 @@ type Config struct {
 	// If not specify any in flags, then genericapiserver will only enable defaultAPIResourceConfig.
 	MergedResourceConfig *serverstore.ResourceConfig
 
+	// AdvertisePort is used when the targetPort and port used in the kubernetes.default.svc service and
+	// endpoint is different than the one used for binding. This configuration is needed in case of a front proxy is
+	// used in front of kube-apiserver
+	AdvertisePort int
```

Comment:
PublicPort to be consistent?

Also though I'm not sure what the "values below here are targets for removal" comment is about, but it looks like PublicPort should be under the same comment as PublicAddress.

nit: missing "." at the end.

Reply:
Looks like the comment regarding values below being targeted for removal was added 6y ago here by @deads2k: https://github.com/kubernetes/kubernetes/pull/38123/files#diff-f1637a4c98f03ff4d53ade7da2226d758a594c9822efb88091aef6afe810f497R154 - not sure why exactly they are expected to be removed though from the associated PR 👀